docs/specs/alcproxies.md

```diff
@@ -0,0 +1,251 @@
+
+A `DispatchProxy` can be used to create a "fake" object of a given type, which acts as if it was the instance we wanted to use. Having `ProxyObject` inherit from `DispatchProxy` allows us to easily add more extra actions when we call the object, such as deserialization of objects across the type barrier (See the "Type-Isolation Barrier" section below).
+
+This design is dependent on the ability for the `ProxyBuilder` to create dynamic assemblies and types within a different ALC/Process. Currently we may be able to do so by loading the `DispatchBuilder` type into the ALC directly, and then creating the proxy object from there, while also using a similar builder to create the user-side proxy that make gRPC calls. We can also wait (See [#38426](https://github.com/dotnet/corefx/issues/38426 )), but this option will not be available until said issue can be fixed, or a comparable fix is made.
+
+### Variant for ALC Revisions with `DispatchProxy`
+
+![](../img/ALCProxyAlternate.svg)
+
+One thing that's nice about `DispatchProxy` is that it allows for us to load a target Type directly from the target ALC to the user ALC, allowing us to remove the `ServerObject`, and instead just call the object's methods directly. This only works for in-process communications, though, and by removing `ServerObject`, we may lose the ability to allow for consistent unloadability (due to how the `TargetObject` may return objects), so this approach needs to be investigated further.
+
+
+### The Type-Isolation barrier, and Client/Server send options
+One of the problems with getting proxies like this working is getting past the type-isolation barrier from ALCs. For example, if you want to use type `Foo`, loading `Foo` directly vs using an ALC generates what the runtime says are completely different types, even if the implementation is the same. 
+
+![](../img/ALCContextError.PNG)
+
+This happens for any non-simple types that attempt to cross the barrier, which is why API needs to use reflection to get the type of the ProxyObject instead of being able to grab the equivalent type in the user context.
+
+The type-isolation barrier becomes a larger issue for parameters and return types in particular, as when we pass a `Foo` type into the proxy, we need to transform it into the `ALC.Foo` type during the transition. There are a few ways we could do this:
+* Build proxies for each non-simple type that can communicate across the ALC boundaries
+* Make everything a shared type (This is not recommended)
```

Comment:
Not even a valid option.

Reply:
That depends what this statement means - if we would change it to "Make everything which is used to communicate between ALCs a shared type" then I think that's a very valid option - and in many ways probably the preferred solution.